docs/infrastructure.md

```diff
@@ -223,36 +223,87 @@ curl -X PUT -i -u "admin:NewPassword" http://localhost:5820/admin/permissions/us
+we can add some example graph-ready data to the new graph database.
+
+First, clone the [`neurobagel_examples`](https://github.com/neurobagel/examples) repository:
 
 ```bash
 git clone https://github.com/neurobagel/neurobagel_examples.git
-
-# Upload the two example files to the graph
-for example in neurobagel_examples/*.ttl; do
-    curl -u "admin:NewPassword" -i -X POST http://localhost:5820/test_data \
-    -H "Content-Type: text/turtle" \
-    --data-binary @{example}
-done
 ```
 
+Next, upload the `.jsonld` file in the directory `neurobagel_examples/data-upload/pheno-bids-output` to the database we created above, using `add_data_to_graph.sh`:
+
 !!! info
     Normally you would create the graph-ready files by first [annotating
     the phenotypic information of a BIDS dataset](../annotation_tool) with the 
     Neurobagel annotator, and then [parsing the annotated BIDS
     dataset](../cli) with the Neurobagel CLI.
 
-### Test the new deployment
+```bash
+./add_data_to_graph.sh PATH/TO/neurobagel_examples/data-upload/pheno-bids-output \ 
+  localhost:7200 repositories/my_db/statements NewUser NewUserPassword \
+  --clear-data
+```
+**Note:** Here we added the `--clear-data` flag to remove any existing data in the database (if the database is empty, the flag has no effect).
+You can choose to omit the flag or explicitly specify `--no-clear-data` (default behaviour) to skip this step.
+
+### Updating a dataset in the graph database
+If the raw data for a previously harmonized dataset (i.e., already has a corresponding JSONLD _which is in the graph_) has been updated, [a new JSONLD file must first be generated for that dataset](updating_dataset.md).
+To push the update to the corresponding graph database, our current recommended approach is to simply clear the database and re-upload all existing datasets, including the **new** JSONLD file for the updated dataset.
+
+To do this, rerun `add_data_to_graph.sh` on the directory containing the JSONLD files currently in the graph database, including the replaced JSONLD file for the dataset that has been updated.
+**Make sure to include the `--clear-data` flag when running the script so that the database is cleared first.**
+
+## Where to store Neurobagel graph-ready data
+To allow easy (re-)uploading of datasets when needed, we recommend having a shared directory in your data filesystem/server for storing Neurobagel graph-ready JSONLD files created for datasets at your institute or lab. 
+This directory can be called anything you like, but we recommend an explicit name such as `neurobagel_jsonld_datasets` to distinguish it from the actual raw data files or Neurobagel data dictionaries.
+Each `.jsonld` in the directory should include the name of the dataset in the filename.
+
+If you anticipate having to add _new_ datasets to a created graph database in the future, we recommend having a separate directory for JSONLD files of datasets that are already in the graph vs. to be added to the graph, e.g.:
```

Comment:
Basically, to minimize unnecessarily wiping the existing graph database where possible. My thought process was that of the two possibilities for more data entering the graph:
a. An existing dataset receives an update
b. A completely new dataset finishes collection + annotation and now has a JSONLD ready for upload

Only in a) would you need to clear the existing graph, since b) would not affect any of the existing datasets and so could just be directly **added** to the existing graph, which is also faster. Since it's possible that there are time intervals between when sysadmins update their graph database, b) may also include multiple new datasets. In this scenario, I think it helps to track which datasets have not been uploaded yet vs. ones that have received an update in the interim. If it's only the latter, the sysadmin wouldn't need to clear the database first and can just upload all the new JSONLD files from the second directory.

Does that make sense?

Reply:
yeah, I get it. I would still stick to the "dumb but clear" suggestion for clarity's sake. New user just looking for reasonable defaults will be OK to do a bit extra work (wipe graph on every change) in exchange for easy to understand workflows. Veteran user will likely do what they think is best regardless of what you write here. So I'd opt for maximum clarity.